src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java

```diff
@@ -61,7 +63,20 @@ public void publish(final ChecksDetails details) {
                     credentials);
 
             GitHubChecksDetails gitHubDetails = new GitHubChecksDetails(details);
-            createBuilder(gitHub, gitHubDetails).create();
+
+            Optional<Long> existingId = context.getId(gitHubDetails.getName());
+
+            final GHCheckRun run;
+
+            if (existingId.isPresent()) {
+                run = createUpdater(gitHub, gitHubDetails, existingId.get()).create();
+            }
+            else {
```

Comment:
Having to do this made me a bit queasy. I've not seen it the (admittedly small sample of) other Jenkins plugins I've seen...

Reply:
The style is borrowed from [warnings-ng plugin](https://github.com/jenkinsci/warnings-ng-plugin) which is an amazing plugin with good quality, it's the first plugin that I contributed, so I just used some of the configurations from that.

Reply:
I'm sure I'll cope with the extra newlines :)

Reply:
this instance is weird as far as I know this violates the standard java style guides for everything, but didn't want to fight this when I touched it

Reply:
actually, it's not my normal style as well , but since it has already started, better to stick to it 😅

Or maybe I can try to overwrite this single rule in the pom, no idea if that's feasible, however.